quickstarts/microsoft.machinelearningservices/machine-learning-workspace-cmk/azuredeploy.json

```diff
@@ -0,0 +1,191 @@
+      "cmk_keyvault_id": {
+        "type": "string",
+        "defaultValue": "/subscriptions/<subscription>/resourceGroups/<resourcegroup>/providers/Microsoft.KeyVault/vaults/<vault>",
+        "metadata": {
+          "description": "Specifies the customer managed keyvault Resource Manager ID."
+        }
+      },
+      "cmk_keyvault_key_uri":{
+        "type": "string",
+        "defaultValue": "https://<vault>.vault.azure.<cloud>/keys/<name>/<version>",
+        "metadata": {
+          "description": "Specifies the customer managed keyvault key uri."
+        }
+      }
+  },
+  "variables": {
+      "tenantId": "[subscription().tenantId]",
+      "storageAccountName": "[concat('st', parameters('name'), parameters('environment'))]",
+      "keyVaultName": "[concat('kv-', parameters('name'), '-', parameters('environment'))]",
+      "applicationInsightsName": "[concat('appi-', parameters('name'), '-', parameters('environment'))]",
+      "resourceGroupName": "[resourceGroup().name]",
+      "containerRegistryName": "[concat('cr', parameters('name'), parameters('environment'))]",
+      "workspaceName": "[concat('mlw', parameters('name'), parameters('environment'))]",
+      "storageAccount": "[resourceId(variables('resourceGroupName'), 'Microsoft.Storage/storageAccounts', variables('storageAccountName'))]",
+      "keyVault": "[resourceId(variables('resourceGroupName'), 'Microsoft.KeyVault/vaults', variables('keyVaultName'))]",
+      "applicationInsights": "[resourceId(variables('resourceGroupName'), 'Microsoft.Insights/components', variables('applicationInsightsName'))]",
+      "containerRegistry": "[resourceId(variables('resourceGroupName'), 'Microsoft.ContainerRegistry/registries', variables('containerRegistryName'))]"
+  },
+  "resources": [
+      {
+          "type": "Microsoft.Storage/storageAccounts",
+          "apiVersion": "2021-01-01",
+          "name": "[variables('storageAccountName')]",
+          "location": "[parameters('location')]",
+          "sku": {
+              "name": "Standard_RAGRS"
+          },
+          "kind": "StorageV2",
+          "properties": {
+              "encryption": {
+                  "services": {
+                      "blob": {
+                          "enabled": true
+                      },
+                      "file": {
+                          "enabled": true
+                      }
+                  },
+                  "keySource": "Microsoft.Storage"
+              },
+              "supportsHttpsTrafficOnly": true
+          }
+      },
+      {
+          "type": "Microsoft.KeyVault/vaults",
+          "apiVersion": "2021-04-01-preview",
+          "name": "[variables('keyVaultName')]",
+          "location": "[parameters('location')]",
+          "properties": {
+              "tenantId": "[variables('tenantId')]",
+              "sku": {
+                  "name": "standard",
+                  "family": "A"
+              },
+              "accessPolicies": [],
```

Comment:
How is this used? (no one has access)

Reply:
Azure ML assigns these later, but indeed might as well not specify it.

Reply:
This needs to be added as required property. Azure ML will add the access policies during provisioning.

Reply:
It is required... how is the vault actually used - since there is a prereq for the encryption key?

Reply:
Azure ML takes Key Vault as a dependency for secret management. At time of resource provisioning, an RBAC role assignment is created for the Workspace MSI on the AKV resource and access policies will be created accordingly. You can find more details here: https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace#resources